hooks/helpers/useHandleError.ts

```diff
@@ -1,16 +1,48 @@
+import { errorFetchingVoteDataMessage, wrongChainMessage } from "constant";
+import { formatTransactionError } from "helpers";
 import { useErrorContext } from "hooks";
 import { ErrorOriginT } from "types";
 
-export function useHandleError(errorOrigin?: ErrorOriginT) {
-  const { addErrorMessage } = useErrorContext(errorOrigin);
+interface Options {
+  errorOrigin?: ErrorOriginT;
+  isContractTransaction?: boolean;
+  isDataFetching?: boolean;
+  customErrorMessage?: string;
+}
+export function useHandleError(options?: Options) {
+  const {
+    errorOrigin,
+    isContractTransaction = true,
+    isDataFetching = false,
+    customErrorMessage,
+  } = options || {};
+  const { addErrorMessage, clearErrorMessages, errorMessages } =
+    useErrorContext(errorOrigin);
+
+  function onError(error: unknown) {
+    if (errorMessages.includes(wrongChainMessage)) return;
```

Comment:
If the user is on the wrong chain, we might as well not show any other errors. Nothing will work until they get onto the correct chain.